protocol/x/affiliates/keeper/keeper.go

```diff
@@ -162,41 +179,43 @@ func (k Keeper) GetTierForAffiliate(
 	numTiers := uint32(len(affiliateTiers.GetTiers()))
 	maxTierLevel := numTiers - 1
 	currentTier := uint32(0)
-	referredVolume, exists := k.GetReferredVolume(ctx, affiliateAddr)
-
-	if !exists {
-		// If referred volume is not found, set it to 0.
-		referredVolume = dtypes.NewInt(0)
+	referredVolume, err := k.GetReferredVolume(ctx, affiliateAddr)
+	if err != nil {
+		return 0, 0, err
 	}
 
-	for _, tier := range affiliateTiers.GetTiers() {
-		if referredVolume.BigInt().Int64() >= int64(tier.ReqReferredVolume) {
-			currentTier = tier.GetLevel()
+	for index, tier := range affiliateTiers.GetTiers() {
+		if referredVolume.Int64() >= int64(tier.ReqReferredVolume) {
```

Comment:
Can't `referredVolume.Int64()` overflow? Seems we should convert `tier.ReqReferredVolume` for big int comparison instead